sycl/include/CL/sycl/ONEAPI/reduction.hpp

```diff
@@ -359,6 +388,10 @@ class reduction_impl {
   static constexpr access::mode accessor_mode = AccMode;
   static constexpr int accessor_dim = Dims;
   static constexpr int buffer_dim = (Dims == 0) ? 1 : Dims;
+  using local_accessor_type =
+      accessor<T, buffer_dim, access::mode::discard_read_write,
```

Comment:
```suggestion
      accessor<T, buffer_dim, access::mode::read_write,
```

Table 4.48 of SYCL 1.2.1 (Table 4.55 in SYCL 2020 provisional) says local accessors only support `read_write` and `atomic` access modes.